torch/distributed/_tensor/op_schema.py

```diff
@@ -54,17 +54,49 @@ def __str__(self) -> str:
         return f"({input_specs_str}) -> ({output_spec_str}) @ mesh layout: {tuple(self.output_spec.mesh.mesh.shape)}"
 
 
-@dataclass
-class StrategyList(object):
+class StrategyType(object):
     """
-    List of placement strategies associated with an op
+    Base class type for op strategy, We have two StrategyType:
+        OpStrategy and TupleStrategy
     """
 
-    strategies: List[PlacementStrategy]
+    pass
+
+
+class OpStrategy(StrategyType):
+    """
+    OpStrategy that consists of a list of placement strategies associated with the op
+    """
+
+    def __init__(self, strategies: List[PlacementStrategy]) -> None:
+        super().__init__()
+        self.strategies = strategies
 
     def __str__(self) -> str:
         strategy_list_str = ", ".join([str(strategy) for strategy in self.strategies])
-        return f"StrategyList: [{strategy_list_str}]"
+        return f"OpStrategy: [{strategy_list_str}]"
+
+
+class TupleStrategy(StrategyType):
+    """
+    TupleStrategy represents the output strategy of this op is a tuple
+    of strategy lists, this would be used if the output strategies might
```

Comment:
do we always return a TupleStrategy if the output is a tuple of tensors? Or do we only use this when those tensors have different strategies?

Reply:
This is a good point, let me think through this a bit and see if we should consolidate on always returning TupleStrategy

Reply:
for `List[Tensor]` return I think we should have a single OpStrategy, where each elements share the same output spec, but for `Tuple[Tensor, ..]` I think we should return TupleStrategy regardless of whether the strategies are same or not, I'll update the comment.